relay-metrics/src/protocol.rs

```diff
@@ -45,12 +204,20 @@ impl fmt::Display for DurationPrecision {
 /// Units and their precisions are uniquely represented by a string identifier.
 #[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
 pub enum MetricUnit {
-    /// A time duration, defaulting to milliseconds (`"ms"`).
-    Duration(DurationPrecision),
+    /// A time duration, defaulting to milliseconds (`"millisecond"`).
+    Duration(DurationUnit),
+    /// TODO: Doc
+    Information(InformationUnit),
+    /// TODO: Doc
+    Fraction(FractionUnit),
+    /// TODO: Doc
+    Custom(CustomUnit),
     /// Untyped value without a unit (`""`).
     None,
 }
 
+// TODO: Test sizeof and alignof
```

Comment:
The idea behind making `CustomUnit` a `[u8;15]` is so that `sizeof(MetricUnit) == 16`. This should be asserted with a test, and documented on `CustomUnit`.